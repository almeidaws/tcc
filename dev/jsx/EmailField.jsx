import Field from './Field.jsx';

const EmailField = props => <Field 
                                type="email" 
                                placeholder="Enter Your Email" 
                                icon="envelope" 
                                required={true}
                                errorMessage={props.errorMessage}
                                value={props.value} 
                                onChange={props.onChange} />;
export default EmailField;