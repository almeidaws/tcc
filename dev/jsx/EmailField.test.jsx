import React from 'react';
import EmailField from './EmailField.jsx';
import renderer from 'react-test-renderer';

it('<EmailField /> should render email', () => {
    const component = renderer.create(<EmailField
                                        value="email.teste@gmail.com"
                                        />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

it('<EmailField /> should present an error', () => {
    const component = renderer.create(<EmailField
                                        value="asds/"
                                        />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});