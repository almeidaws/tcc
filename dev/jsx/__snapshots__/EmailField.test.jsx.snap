// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmailField /> should present an error 1`] = `
<div
  className="form-group"
>
  <input
    className="form-control"
    placeholder="Enter Your Email"
    required={true}
    type="email"
    value="asds/"
  />
  <span
    className="form_icon"
  >
    <i
      aria-hidden="true"
      className="fa_icon form-envelope"
    />
  </span>
  <div
    className="invalid-feedback text-warning"
  />
</div>
`;

exports[`<EmailField /> should render email 1`] = `
<div
  className="form-group"
>
  <input
    className="form-control"
    placeholder="Enter Your Email"
    required={true}
    type="email"
    value="email.teste@gmail.com"
  />
  <span
    className="form_icon"
  >
    <i
      aria-hidden="true"
      className="fa_icon form-envelope"
    />
  </span>
  <div
    className="invalid-feedback text-warning"
  />
</div>
`;
